c2corg_api/views/document_revert.py

```diff
@@ -0,0 +1,236 @@
+            return Route, RouteLocale, ArchiveRoute, ArchiveRouteLocale, \
+                   document_type
+        if document_type == OUTING_TYPE:
+            return Outing, OutingLocale, ArchiveOuting, ArchiveOutingLocale, \
+                   document_type
+        if document_type == IMAGE_TYPE:
+            return Image, None, ArchiveImage, ArchiveDocumentLocale, \
+                   document_type
+        if document_type == ARTICLE_TYPE:
+            return Article, None, ArchiveArticle, ArchiveDocumentLocale, \
+                   document_type
+        if document_type == BOOK_TYPE:
+            return Book, None, ArchiveBook, ArchiveDocumentLocale, \
+                   document_type
+        if document_type == XREPORT_TYPE:
+            return Xreport, XreportLocale, ArchiveXreport, \
+                   ArchiveXreportLocale, document_type
+        if document_type == AREA_TYPE:
+            return Area, None, ArchiveArea, ArchiveDocumentLocale, \
+                   document_type
+        assert False
+
+    def _get_archive_document(self, document_id, lang, version_id,
+                              archive_clazz, archive_locale_clazz):
+        version = DBSession.query(DocumentVersion) \
+            .options(joinedload(
+                DocumentVersion.document_archive.of_type(archive_clazz))) \
+            .options(joinedload(
+                DocumentVersion.document_locales_archive.of_type(
+                    archive_locale_clazz))) \
+            .options(joinedload(DocumentVersion.document_geometry_archive)) \
+            .filter(DocumentVersion.id == version_id) \
+            .filter(DocumentVersion.document_id == document_id) \
+            .filter(DocumentVersion.lang == lang) \
+            .first()
+
+        archive_document = version.document_archive
+        archive_document.geometry = version.document_geometry_archive
+        archive_document.locales = [version.document_locales_archive]
+        return archive_document
+
+    def _get_current_document(self, document_id, lang, clazz, clazz_locale):
+        locales_type = with_polymorphic(DocumentLocale, clazz_locale) \
+            if clazz_locale else DocumentLocale
+        locales_attr = getattr(clazz, 'locales')
+        locales_type_eager = locales_attr.of_type(clazz_locale) \
+            if clazz_locale else locales_attr
+
+        document_query = DBSession. \
+            query(clazz). \
+            join(locales_type). \
+            filter(getattr(clazz, 'document_id') == document_id). \
+            filter(DocumentLocale.lang == lang). \
+            options(joinedload('geometry')). \
+            options(contains_eager(locales_type_eager, alias=locales_type))
+        return document_query.first()
+
+    def _get_after_update(self, document_type):
+        if document_type == WAYPOINT_TYPE:
+            return update_linked_route_titles
+        if document_type == ROUTE_TYPE:
+            return update_title_prefix
```

Comment:
Does not this handle the routes main waypoint title_prefix that you're saying is not handled ?